.github/workflows/scripts/check-files.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/sh
+set -eu
+
+SIZE_LIMIT=150000
+
+check_size() {
+	size="$(stat --printf="%s" "$1")"
+	if [ "$size" -gt "$SIZE_LIMIT" ]; then
+		echo "File $1 is bigger than specified $SIZE_LIMIT limit"
+		exit 1
+	fi
+}
+
+check_webp_name() {
+	shouldname="$(echo "$1" |
+		iconv --to-code=utf-8 |
+		sed "s/^\(data\/pix\/\)\?/data\/pix\//" |
+		tr '[:upper:]' '[:lower:]' |
+		tr '_ ' '-')"
+	if [ "$shouldname" != "$1" ]; then
+		echo "$1 should be named $shouldname."
+		exit 1
+	fi
+}
+
+check_recipe_name() {
+	shouldname="$(echo "$1" |
+		iconv --to-code=utf-8 |
+		sed "s/^\(src\/\)\?/src\//" |
+		tr '[:upper:]' '[:lower:]' |
+		tr '_ ' '-')"
+	if [ "$shouldname" != "$1" ]; then
+		echo "$1 should be named $shouldname."
+		exit 1
+	fi
+}
+
+check_recipe_content() {
+	awk '
+		BEGIN {
+			HAS_TITLE       = 0;
+			HAS_TAGS        = 0;
+			NUM_TAGS        = 0;
+			HAS_INGREDIENTS = 0;
+			HAS_DIRECTIONS  = 0;
+		}
+
+		# First line should be the title
+		NR == 1 && /^# / {
+			HAS_TITLE = 1;
+			next;
+		}
+
+		/^## Ingredients/ {
+			HAS_INGREDIENTS = 1;
+			next;
+		}
+
+		/^## Directions/ {
+			HAS_DIRECTIONS = 1;
+			next;
+		}
+
+		END {
+			# Last line should be the tags list
+			if ($1 == ";tags:") {
+				HAS_TAGS = 1;
+				NUM_TAGS = NF - 1;
+			}
+
+			FAIL = 0;
+
+			if (!HAS_TITLE) {
+				print "Recipe does not have a properly formatted title on the first line."
+				FAIL = 1;
+			}
+
+			if (!HAS_TAGS) {
+				print "Recipe does not have a properly formatted tags on the last line."
+				FAIL = 1;
+			} else if (NUM_TAGS < 2) {
+				print "Recipe only has " NUM_TAGS " tags. Add some more."
+				FAIL = 1;
+			} else if (NUM_TAGS > 5) {
+				print "Recipe has " NUM_TAGS " tags which is too many. Remove some tags."
+				FAIL = 1;
+			}
+
+			if (!HAS_INGREDIENTS) {
+				print "Recipe does not have an ingredients list."
+				FAIL = 1;
+			}
+
+			if (!HAS_DIRECTIONS) {
+				print "Recipe does not have a directions section."
+				FAIL = 1;
+			}
+
+			if (FAIL) {
+				exit 1;
+			}
+		}
+	' "$1"
+
+	if [ $? -ne 0 ]; then
+		exit 1;
+	fi
+}
+
+git diff --name-only "$(git merge-base origin/master HEAD)" | while IFS= read -r file; do
+	case "$file" in
+		*.webp)
+			echo "Checking size of $file"
+			check_size "$file"
+			check_webp_name "$file"
+			;;
+		.github/*.md)
+			exit 0;
```

Comment:
Looks like it will exit as soon as it meets .md file in .github subdirectory leaving other files uncheked

Reply:
This is the intent. The markdown file in the .github folder shouldn't be checked because the markdown checks in this file expect a recipe file.

Reply:
Imagine you have a list of files in the diff, f.e: `foo.md`, `.github/baz.md`, `bar.webp`. Script will exit with 0 exit code after it meets the second file and `bar.webp` will not be checked

Reply:
woops. Thanks for catching that. This has been merged now though, so I'll have to open up a new PR

Reply:
```sh
#!/bin/sh

echo "foo.md\n.github/baz.md\nbar.webp" | while IFS= read -r file; do
    case "$file" in
        *.webp)
            echo "WEBP file $file"
            ;;
        .github/*.md)
            echo "MD in .github $file"
            exit 0
            ;;
        *.md)
            echo "MD to check $file"
            ;;
    esac
done
```

Take a look at this example. Try to run it and then delete `exit 0` part and run again

@LukeSmithxyz you are hurring to much